### PR TITLE
Fixed #24890 -- Added warning to collectstatic when static files have clashing names

### PR DESCRIPTION
**Pull request by alimony**
See ticket: https://code.djangoproject.com/ticket/24890

### CODE REVIEW

File: tests/staticfiles_tests/tests.py [hunk 1]
```diff
@@ -399,6 +399,71 @@ def test_ordering_override(self):
...
+class TestCollectionOverwriteWarning(CollectionTestCase):
+    """
+    Test warning in ``collectstatic`` output when a file is skipped because a
+    previous and separate file was already written to the same path. See
+    Command.collect in contrib/staticfiles/management/commands/collectstatic.py
+    """
+    def setUp(self):
+        # If this string is in the collectstatic output, it means the warning
+        # we're looking for was emitted.
+        self.warning_string = 'Found another file'
```
Comment (timgraham) [inline tests/staticfiles_tests/tests.py]: this variable can be at the class level, no need to set it up for each test method.

File: tests/staticfiles_tests/tests.py [hunk 2]
```diff
@@ -399,6 +399,71 @@ def test_ordering_override(self):
...
+    def test_warning(self):
+        """
+        Make sure the warning is emitted when there are duplicate destinations.
+        """
+        # Create new file in the no_label app that also exists in the test app.
+        test_dir = os.path.join(TEST_ROOT, 'apps', 'no_label', 'static', 'test')
+        if not os.path.exists(test_dir):
+            os.mkdir(test_dir)
+
+        duplicate_path = os.path.join(test_dir, 'file.txt')
```
Comment (timgraham) [inline tests/staticfiles_tests/tests.py]: This should probably be: 
```
try:
    duplicate_path = os.path.join(test_dir, 'file.txt')
    ....
finally:
    if os.path.exists(duplicate_path):
```
so if something fails, we still cleanup. Alternatively, could refactor to use setUp/tearDown methods.

**Comment by timgraham** [comment]
Feel free to squash commits when updating, thanks!

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
Oops, test are not passing (ignore servers failure, that was present on master).

**Comment by alimony** [comment]
Yeah, had not run them properly on Python 2. Needed to add `force_text()` to all output comparisons, as do all other tests in the same file already. Pushed this change just now.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
merged in 5304494585c58b0c9245ea9896a6d6122a8673a2, thanks!

**Pull request closed without merging**